passkit/lib/src/passkit/pkpass.dart

```diff
@@ -111,8 +91,29 @@ class PkPass {
       pass: PassData.fromJson(passJson!),
       manifest: manifestJson!,
       logo: logo,
+      icon: icon,
       footer: footer,
+      thumbnail: thumbnail,
       sourceData: bytes,
+      strip: strip,
+      background: background,
+    );
+  }
+
+  static PkPassImage? _loadImage(Archive archive, String name) {
```

Comment:
maybe this can be an Archive extension method like `loadPassImage({required String filename})` so we can test it (static methods are difficult to test)

Reply:
Honestly, I wouldn't bother testing individual methods in this library. At the end of the day, you can write a test for `PkPass.fromBytes()` and assert that each property matches what you expect. That's also the only API that actually matters for the consumer of the app. Testing the lib that way gives you a lot of flexibility for changing the internal structure.

All the tests you need are basically:
```dart
test('Can read logo from pass file', () async {
  final bytes = await File('test/sample_passes/pass_xyz.pkpass').readAsBytes();

  final pkPass = PkPass.fromBytes(bytes);
  expect(pkPass.logo?.fromMultiplier(1), isNotNull); // or assert whatever else
});
```

Reply:
Another reason not to make it an extension method on archive and testing it individually is that that keeps the `archive` dependency an implementation detail, even in the tests.

Reply:
Oh and another good reason for this "test file goes in - property gets asserted" style of test is that, assuming tests are exhaustive, any uncovered line is actually dead code.